core/metacling/src/TClingCallFunc.cxx

```diff
@@ -528,6 +623,433 @@ void TClingCallFunc::make_narg_ctor_with_return(const unsigned N, const string &
    buf << "}\n";
 }
 
+int TClingCallFunc::generate_wrapper(std::string& wrapper_name, std::string& wrapper) {
+  const FunctionDecl *FD = GetDecl();
```

Comment:
If this is the only class member that we need we could transform this routine into a public static one taking a `DeclId_t`. We will call it from the inside with `GetDecl()->getCanonicalDecl()` see `TCling::GetDeclId` (not really sure why it is not static, we could have reused it).

From the outside, we won't need to have a special method in `TInterpreter` and `TCling` but rely on the static interface.